docs/book/src/migration/guide.md

```diff
@@ -0,0 +1,223 @@
+# Migration from v1 to v2
+
+Make ensure you understand the [differences between Kubebuilder v1 and v2](./v1vsv2.md)
+before continuing
+
+Please ensure you have followed the [installation guide](/quick-start.md#installation)
+to install the required components.
+
+The recommended way to migrate a v1 project is to create a new v2 project and
+copy over the API and the reconciliation code. The conversion will end up with a
+project that looks like a native v2 project. However, in some cases, it's
+possible to do a in-place upgrade. i.e. reuse the v1 project layout and only
+upgrade controller-runtime and controller-tools.  
+
+Let's take the [example v1 project][v1-project] and migrate it to Kubebuilder
+v2. At the end, we should have something that looks like the
+[example v2 project][v2-project].
+
+## Preparation
+
+What are the group, version, kind and domain? We'll need to figure out what the
+group, version, kind and domain are.
+
+Let's take a look at our current v1 project structure:
+
+```
+pkg/
+├── apis
+│   ├── addtoscheme_batch_v1.go
+│   ├── apis.go
+│   └── batch
+│       ├── group.go
+│       └── v1
+│           ├── cronjob_types.go
+│           ├── cronjob_types_test.go
+│           ├── doc.go
+│           ├── register.go
+│           ├── v1_suite_test.go
+│           └── zz_generated.deepcopy.go
+├── controller
+└── webhook
+```
+
+All of our API information is stored in the `pkg/apis/batch`, so we can look
+there to find what we need to know.
+
+In `cronjob_types.go`, we can find
+
+```go
+type CronJob struct {...}
+```
+
+In `register.go`, we can find
+
+```go
+SchemeGroupVersion = schema.GroupVersion{Group: "batch.tutorial.kubebuilder.io", Version: "v1"}
+```
+
+Now we have figured out what the group, version, kind and domain are.
+
+- domain: tutorial.kubebuilder.io
+- group: batch
+- version: v1
+- kind: CronJob
+
+## Initialize a v2 Project
+
+Now that we initialize a v2 project, we'll need to first initialize the go
+modules.
+
+```bash
+go mod init tutorial.kubebuilder.io/project
+```
+
+Then we finish initializing the project with kubebuilder.
```

Comment:
while gramatically correct, this is a bit of an awkward construction -- it's generally only used in cooking shows, painting tutorials, etc.

I'd use

```suggestion
Then, we can finish initializing the project with kubebuilder:
```

or just

```suggestion
Then, we can initialize the project with kubebuilder:
```